File: day9/go/point.go
package main

type Point struct {
	row int
	col int
}

// NewPoint creates a new Point with the specified row and column.
func NewPoint(row, col int) *Point {
	return &Point{row, col}
}

// Touches returns true if the square of the Euclidean distance between
// this point and that point is less than or equal to 2.
func (this *Point) Touches(that *Point) bool {
	rowdif := this.row - that.row
	coldif := this.col - that.col
	return rowdif*rowdif+coldif*coldif <= 2
}

File: day9/point.py
class Point:
    def __init__(self, row, col):
        self.row = row
        self.col = col

    def touches(self, other):
        rowdif = self.row - other.row
        coldif = self.col - other.col
        return rowdif*rowdif + coldif*coldif <= 2
    
    def __eq__(self, other):
        return self.row == other.row and self.col == other.col
